### PR TITLE
LLM calculator documentation updates

### PR DESCRIPTION
**Pull request by dtrawins**
### 🛠 Summary

CVS-151888 CVS-151890
LLM calculator documentation updates.

### 🧪 Checklist

- [ ] Unit tests added.
- [x] The documentation updated.
- [ ] Change follows security best practices.
``

### CODE REVIEW

File: docs/llm/reference.md [hunk 1]
```diff
@@ -96,8 +94,29 @@ You can track the actual usage of the cache in the server logs. You can observe
 ```
 Consider increasing the `cache_size` parameter in case the logs report the usage getting close to 100%. When the cache is consumed, some of the running requests might be preempted to free cache for other requests to finish their generations (preemption will likely have negative impact on performance since preempted request cache will need to be recomputed when it gets processed again). When preemption is not possible i.e. `cache size` is very small and there is a single, long running request that consumes it all, then the request gets terminated when no more cache can be assigned to it, even before reaching stopping criteria. 
 
+`enable_prefix_caching` can improve generation performance when the initial prompt content is repeated. That is the case in chat application which resend the history on the conversations. Thanks to prefix caching, there is not need to reevaluate the same sequence of tokens. Thanks to that, first token will be generated much quicker and the overall
```
Comment (dkalinowski) [inline docs/llm/reference.md:97]:
```suggestion
`enable_prefix_caching` can improve generation performance when the initial prompt content is repeated. That is the case with chat applications which resend the history of the conversations. Thanks to prefix caching, there is no need to reevaluate the same sequence of tokens. Thanks to that, first token will be generated much quicker and the overall
```

File: docs/llm/reference.md [hunk 2]
```diff
@@ -96,8 +94,29 @@ You can track the actual usage of the cache in the server logs. You can observe
...
+
+`plugin_config` accepts a json dictionary of tuning parameters for the OpenVINO plugin. It can tune the behavior of the inference runtime. For example you can include there kv cache compression or the group size '{"KV_CACHE_PRECISION": "u8", "DYNAMIC_QUANTIZATION_GROUP_SIZE": "32"}'.
+
 The LLM calculator config can also restrict the range of sampling parameters in the client requests. If needed change the default values for  `max_tokens_limit` and `best_of_limit`. It is meant to avoid the result of memory overconsumption by invalid requests.
 
+
+## Canceling the generation
+
+In order to optimize the usage of compute resources, it is important to stop the text generation when it became irrelevant for the client or when the client gets disconnected for any reason. Such capability is implemented via a tight integration between the LLM calculator and the model server frontend. The calculator gets notified about the client session disconnection. When the client application stops or deliberately breaks the session, the generation cycle gets broken and all resources are released. Below is an easy example how the client can initialize stopping the generation:
```
Comment (dkalinowski) [inline docs/llm/reference.md:107]:
```suggestion
In order to optimize the usage of compute resources, it is important to stop the text generation when it becomes irrelevant for the client or when the client gets disconnected for any reason. Such capability is implemented via a tight integration between the LLM calculator and the model server frontend. The calculator gets notified about the client session disconnection. When the client application stops or deliberately breaks the session, the generation cycle gets broken and all resources are released. Below is an easy example how the client can initialize stopping the generation:
```

File: docs/llm/reference.md [hunk 3]
```diff
@@ -164,11 +183,11 @@ When default template is loaded, servable accepts `/chat/completions` calls when
 
 ## Limitations
 
-LLM calculator is a preview feature. It runs a set of accuracy, stability and performance tests, but the next releases targets production grade quality. It has now a set of known issues:
+There are several known limitations which are expected to be addressed in the coming releases:
 
 - Metrics related to text generation are not exposed via `metrics` endpoint. Key metrics from LLM calculators are included in the server logs with information about active requests, scheduled for text generation and KV Cache usage. It is possible to track in the metrics the number of active generation requests using metric called `ovms_graphs_running`. Also tracking statistics for request and responses is possible. [Learn more](../metrics.md)
```
Comment (dkalinowski) [inline docs/llm/reference.md:188]:
```suggestion
- Metrics related to text generation are not exposed via `metrics` endpoint. Key metrics from LLM calculators are included in the server logs with information about active requests, scheduled for text generation and KV Cache usage. It is possible to track in the metrics the number of active generation requests using metric called `ovms_current_graphs`. Also tracking statistics for request and responses is possible. [Learn more](../metrics.md) 
```

**Review by dkalinowski — APPROVED** [review]

File: demos/continuous_batching/README.md [hunk 4]
```diff
@@ -299,9 +302,9 @@ Check the example in the [RAG notebook](https://github.com/openvinotoolkit/model
 
 ## Scaling the Model Server
 
-Check this simple [text generation scaling demo](./scaling/README.md).
+Check this simple [text generation scaling demo](https://github.com/openvinotoolkit/model_server/blob/main/demos/continuous_batching/scaling/README.md).
```
Comment (atobiszei) [inline demos/continuous_batching/README.md:305]: Why can't we have relative link here?

File: docs/llm/quickstart.md [hunk 5]
```diff
@@ -4,8 +4,8 @@ Let's deploy [TinyLlama/TinyLlama-1.1B-Chat-v1.0](https://huggingface.co/TinyLla
 
 1. Install python dependencies for the conversion script:
 ```bash
-export PIP_EXTRA_INDEX_URL="https://download.pytorch.org/whl/cpu"
-pip3 install "optimum-intel[nncf,openvino]"@git+https://github.com/huggingface/optimum-intel.git@xeon openvino-tokenizers transformers==4.41.2
+export PIP_EXTRA_INDEX_URL="https://download.pytorch.org/whl/cpu https://storage.openvinotoolkit.org/simple/wheels/pre-release"
+pip3 install --pre "optimum-intel[nncf,openvino]"@git+https://github.com/huggingface/optimum-intel.git  openvino_tokenizers==2024.4.* openvino==2024.4.*
```
Comment (atobiszei) [inline docs/llm/quickstart.md:8]: Why are we adding here pre-release packages?

Comment (dtrawins) [reply docs/llm/quickstart.md:8]: they will be swapped with release packages after their release

File: demos/continuous_batching/README.md (hunk 4 above)
Comment (dtrawins) [reply demos/continuous_batching/README.md:305]: to make it accessible from docs.openvino.ai

**Review by atobiszei — APPROVED** [review]

**Pull request merged**